File: Python/Math-Array-String/49.py
# 2114. Maximum Number of Words Found in Sentences

from typing import List

class Solution:
    def mostWordsFound(self, sentences: List[str]) -> int:
        return max(len(sentence.split()) for sentence in sentences)
    

sol = Solution()
output = sol.mostWordsFound(["alice and bob love leetcode", "i think so too", "this is great thanks very much"])
print(output) # Output: 6

File: JavaScript/Math-Array-String/49.js
// 2114. Maximum Number of Words Found in Sentences


/**
 * @param {string[]} sentences
 * @return {number}
 */
var mostWordsFound = function(sentences) {
    // Use map to count words in each sentence, then find the maximum
    return Math.max(...sentences.map(sentence => sentence.split(' ').length));
};

// Example usage:
const sentences = [
    "alice and bob love leetcode", 
    "i think so too", 
    "this is great thanks very much"
];
console.log(mostWordsFound(sentences)); // Output: 6
